### ISSUE TITLE
Build fails : 'rv64i_zknd_decrypt' undeclared (first use in this function)

### ISSUE THREAD

**Issue by mtuxpe** [openbouffalo/buildroot_bouffalo#83]
I am using Ubuntu Ubuntu 22.04.3 LTS

providers/implementations/ciphers/cipher_aes_xts_hw.c:173:40: error: 'rv64i_zknd_decrypt' undeclared (first use in this function)
  173 |                    rv64i_zkne_encrypt, rv64i_zknd_decrypt,
      |                                        ^~~~~~~~~~~~~~~~~~
providers/implementations/ciphers/cipher_aes_xts_hw.c:29:40: note: in definition of macro 'XTS_SET_KEY_FN'
   29 |         xctx->xts.block1 = (block128_f)fn_block_dec;                           \
      |                                        ^~~~~~~~~~~~
providers/implementations/ciphers/cipher_aes_xts_hw.c: In function 'ossl_prov_cipher_hw_aes_xts':
providers/implementations/ciphers/cipher_aes_xts_hw.c:185:5: warning: implicit declaration of function 'RISCV_HAS_ZKND_AND_ZKNE' [-Wimplicit-function-declaration]
  185 | if (RISCV_HAS_ZKND_AND_ZKNE())                                                 \
      |     ^~~~~~~~~~~~~~~~~~~~~~~
providers/implementations/ciphers/cipher_aes_xts_hw.c:244:5: note: in expansion of macro 'PROV_CIPHER_HW_select_xts'
  244 |     PROV_CIPHER_HW_select_xts()
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~
make[3]: *** [Makefile:21626: providers/implementations/ciphers/libdefault-lib-cipher_aes_xts_hw.o] Error 1
make[2]: *** [Makefile:3045: build_sw] Error 2
make[1]: *** [package/pkg-generic.mk:283: /mnt/ssd/buildroot_bouffalo/buildroot/output/build/libopenssl-3.2.0/.stamp_built] Error 2
make: *** [Makefile:82: _all] Error 2

**Comment by gwnichol** [comment]
I ran into the same issue earlier and have found a temporary solution:
Apply this change to the buildroot repo:
```
diff --git a/package/libopenssl/Config.in b/package/libopenssl/Config.in
index 3199de8cfe..49958d9fd6 100644
--- a/package/libopenssl/Config.in
+++ b/package/libopenssl/Config.in
@@ -23,6 +23,8 @@ config BR2_PACKAGE_LIBOPENSSL_TARGET_ARCH
        default "linux-x86"                     if BR2_i386 && !BR2_TOOLCHAIN_HAS_LIBATOMIC
        default "linux-x86-latomic"             if BR2_i386 && BR2_TOOLCHAIN_HAS_LIBATOMIC
        default "linux-x86_64"                  if BR2_x86_64
+       default "linux64-riscv64"               if BR2_riscv && BR2_ARCH_IS_64
+       default "linux32-riscv32"               if BR2_riscv && !BR2_ARCH_IS_64
        # no-asm is needed with generic architectures such as
        # linux-generic{32,64}, see
        # https://github.com/openssl/openssl/issues/9839
```

This change sets the correct configuration to be used when building libopenssl. Unfortunately buildroot does not currently support setting this option from the defconfig and we previously fall back to `linux-generic64`.

We're only now having build issues due to the definition checks partially addressed in https://github.com/openssl/openssl/issues/22871

**Comment by gwnichol** [comment]
The fix for this in upstream buildroot has been merged in [fc8eff0c76ab35db7d783ce7959193ff4c30a01e](https://git.buildroot.net/buildroot/commit/?id=fc8eff0c76ab35db7d783ce7959193ff4c30a01e).

You should be able to pull down the latest buildroot `master` and build.

**Comment by mtuxpe** [comment]
Got @gwnichol . Many thanks for help.
